core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -945,7 +960,8 @@ private void evaluateTriggerable(Triggerable t, TreeReference anchorRef) {
      * when we support dependent XPath Steps (IE: /path/to//)
      */
     public void addChildrenOfReference(TreeReference original, Vector<TreeReference> toAdd) {
-        for (TreeReference ref : exprEvalContext.expandReference(original)) {
+        Vector<TreeReference> refs = exprEvalContext.expandReference(original);
```

Comment:
`original` is generic and `refs` is the list of possible nodes it can point to. Why? Hmmm, is it that I should be caching only after expanding, not before?